### PR TITLE
chore(dev): Update CODEOWNERS to reflect consolidation

### PR DESCRIPTION
**Pull request by jszwedko**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,100 +1,100 @@
 docs/ @vectordotdev/ux-team
```
Comment (bruceg) [inline .github/CODEOWNERS:1]: Do we still have a `ux-team`?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,100 +1,100 @@
...
+src/sources/redis/ @vectordotdev/vector
+src/sources/socket/ @vectordotdev/vector
+src/sources/splunk_hec/ @vectordotdev/vector
+src/sources/statsd/ @vectordotdev/vector
+src/sources/syslog.rs @vectordotdev/vector
+src/sources/vector/ @vectordotdev/vector
+src/test_util/ @vectordotdev/vector
+src/topology/ @vectordotdev/vector
+src/transforms/ @vectordotdev/vector
 website/ @vectordotdev/ux-team @vectordotdev/documentation
```
Comment (bruceg) [inline .github/CODEOWNERS:100]: Ditto `documentation`

**Review by bruceg — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (jszwedko) [reply .github/CODEOWNERS:1]: Nominally, yes. I'd like to keep this one to have a smaller group continue to build up sufficient context to steer Vector's UX.

File: .github/CODEOWNERS (hunk 2 above)
Comment (jszwedko) [reply .github/CODEOWNERS:100]: This one is still used since there is a separate set of individuals that reviews documentation changes.

**Pull request merged**